### PR TITLE
[Merged by Bors] - Add feature flag to enable wasm for bevy_audio

### PR DESCRIPTION
**Pull request by Ixentus**
Exposes Rodio feature flag to enable WASM support.

Note that mp3 doesn't currently work on wasm.

### CODE REVIEW

File: docs/cargo_features.md
Comment (mockersf) [inline docs/cargo_features.md:34]: As you said mp3 doesn't work, could you mention here which format does?

Comment (Ixentus) [reply docs/cargo_features.md]: Done. They should all work except mp3. Rodio is working on using Symphonia as an alternative backend, which might resolve this.

File: Cargo.toml
Comment (MinerSebas) [inline Cargo.toml:69]: This should also have this disclaimer.
```suggestion
# WASM support for audio (Currently only works with flac, wav and vorbis. Not with mp3)
```

**Review by mockersf — APPROVED** [review]

**Comment by Ixentus** [comment]
Is there anything left to do before merging this? I've been happily listening to audio on web for weeks now :)

**Comment by mockersf** [comment]
it needs another approval 🙂

**Review by NiklasEi — APPROVED** [review]
LGTM, thank you.

**Comment by cart** [comment]
Is there a reason to not enable this by default? Rather than adding another cargo feature, maybe we should just selectively enable the "rodio/wasm-bindgen" when building bevy_audio for wasm?

**Comment by mockersf** [comment]
cargo doesn't support per target feature: https://github.com/rust-lang/cargo/issues/1197

**Comment by cart** [comment]
Ah yeah. Maybe we should start considering a move to "feature resolver v2" now that its stable?

**Comment by mockersf** [comment]
> Ah yeah. Maybe we should start considering a move to "feature resolver v2" now that its stable?

The resolver v2 is not something we can enable for consumers of Bevy, they would have to enable it in their project. We should mention it in all the docs, and it's something additional to do when starting with Bevy...

**Comment by cart** [comment]
Arg thats annoying. I guess we'll just need to wait for Rust 2021 to make it the default.

**Pull request closed without merging**
